crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -69,18 +70,47 @@ pub fn get_scarb_manifest() -> Result<Utf8PathBuf> {
     Ok(path)
 }
 
-pub fn get_scarb_metadata(manifest_path: &Utf8PathBuf) -> Result<scarb_metadata::Metadata> {
+pub fn get_scarb_metadata(
+    manifest_path: &Utf8PathBuf,
+    with_deps: bool,
```

Comment:
like `Option<bool>` ?

Reply:
yes

Reply:
not sure what's the difference between `None` and `false` in that case

Reply:
I'd assume we'd treat `None` as `false` in the code anyways so no point in changing it in my opinion.

Reply:
My reasoning was that we could get rid of writing additional argument `false` everywhere in our current codebase (at the time of writing this comment at least, every place we used this had `false` as argument). 

This is not a crucial change, but I suppose it would be nice to just be able to get scarb metadata with one argument (manifest) and be non breaking in regards to the previous interface.

Reply:
But we would still have to pass `None` instead of `false`.  Like `get_scarb_metadata(metadata, None)`

Reply:
What we can do though, is create two additional functions `get_scarb_metadata_with_deps` and `get_scarb_metadata_without_deps`, which would call `get_scarb_metadata(metadata, with_deps)` under the hood. This should improve readability.

Reply:
I leave it up to you to decide, you can change it the way you've described or as it currently is then